### PR TITLE
Add a deprecation note to k/k/cluster/log-dump directory

### PR DESCRIPTION
**Pull request by tosi3k**
**What type of PR is this?**
/kind cleanup
/kind documentation

**What this PR does / why we need it**:
`log-dump.sh` is dead, long live the `log-dump.sh`.

The note is to inform the customers of this directory that it's deprecated and to encourage them to adjust their jobs to use the test-infra's version of the log dumping script.

**Which issue(s) this PR fixes**:
No issue is opened for that.

**Special notes for your reviewer**:
/sig scalability
/assign @jkaniuk 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/log-dump/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## This directory is deprecated!
+
+We have recently ported the log dumping utility from kubernetes/kubernetes repository to
```
Comment (jkaniuk) [inline cluster/log-dump/README.md:3]: This will stay here for a while.
I propose:
> Log dumping utility was ported ...

File: cluster/log-dump/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+If you require changes to this script, please consider migrating your jobs to use the new
+log dumping mechanism first.
+
+## Migration steps
+
+Currently, `log-dump.sh` file is added to every newly released `kubekins-e2e` image.
+In order to leverage that script, add `USE_KUBEKINS_LOG_DUMPING` environment variable
+to your test job and set its value to `true`.
+
+For the time being, only GCE and GKE providers are supported by the new log dumping mechanism.
```
Comment (jkaniuk) [inline cluster/log-dump/README.md:14]: I would rephrase this to encourage adoption, saying that only GCE and GKE were migrated so far and specifying migrations steps for the rest.

**Comment by mm4tt** [comment]
Shellcheck seems to be failing
```
In ./cluster/log-dump/log-dump.sh line 75:
  local -r dashline=$(printf %100s | tr " " "-")
                             ^---^ SC2183: This format string has 1 variables, but is passed 0 arguments.
```

File: cluster/log-dump/README.md (hunk 1 above)
Comment (mm4tt) [reply cluster/log-dump/README.md:3]: +1, recently is ambiguous in such a context

Comment (tosi3k) [reply cluster/log-dump/README.md]: Done.

**Comment by tosi3k** [comment]
> Shellcheck seems to be failing

It shouldn't anymore ;).

File: cluster/log-dump/README.md (hunk 2 above)
Comment (jkaniuk) [reply cluster/log-dump/README.md:14]: Could you point to more specific place in kubetest?

Also it seems that old log dumping also supported only gce/gke. Could we drop "old" from the first sentence then? Maybe replacing with `are supported by the log-dump mechanism.`?

Comment (tosi3k) [reply cluster/log-dump/README.md]: Done.

**Comment by wojtek-t** [comment]
/approve

Also, given it's only test related changed, adding to milestone
/milestone v1.20

**Comment by BenTheElder** [comment]
Have we thought about how to signal this in e2e.test?

**Comment by tosi3k** [comment]
> Have we thought about how to signal this in e2e.test?

Could you elaborate a little bit?
Currently this would just print a message prompting people to make their jobs use the new mechanism. Do we want to have some additional signaling?

**Comment by pjh** [comment]
Hi @tosi3k , I'm concerned that this PR does not reference any issue # or provide any context on *why* this code was moved to test-infra. It looks https://github.com/kubernetes/test-infra/pull/19461 forked `log-dump.sh` from here into test-infra, then it was moved in https://github.com/kubernetes/test-infra/pull/19675. Our team depends on `log-dump.sh` - would it be possible to create an issue somewhere so that we can subscribe and get updates as the deprecation / migration happens?

Thanks for validating Windows nodes in https://github.com/kubernetes/test-infra/pull/19461. We will look into migrating to the forked version of `log-dump.sh` in test-infra. cc @jingxu97 @YangLu1031

**Comment by mm4tt** [comment]
@pjh, the main reason is to decouple k/k code from the test util code. We don't want to have log dumping (or any test util) living in the k/k because:
* It's hard or impossible to do any changes there during the k8s code freeze.
* The k/k code is released versioned, meaning that we need to backport every improvement / bug-fix to older releases. It creates unnecessary toil

But point taken. There should be an issue, even post-factum, documenting the reason. @tosi3k, could you take care of it?

**Comment by tosi3k** [comment]
Sorry for the late response.

I will open a relevant issue with the post-factum description tomorrow. You can already use the version of `log-dump.sh` from `test-infra` repo by setting `USE_KUBEKINS_LOG_DUMPING` env var to `true` in your job definition leveraging `kubekins-e2e` image.

**Comment by tosi3k** [comment]
@pjh FYI: https://github.com/kubernetes/kubernetes/issues/96915.

**Comment by pjh** [comment]
Thanks, that's extremely helpful!
